src/test/java/dk/cngroup/lentils/repository/ProgressRepositoryTest.java

```diff
@@ -0,0 +1,144 @@
+@SpringBootTest(classes = {LentilsApplication.class, DataConfig.class, ObjectGenerator.class})
+public class ProgressRepositoryTest {
+
+    @Autowired
+    ProgressRepository repository;
+
+    @Autowired
+    TeamRepository teamRepository;
+
+    @Autowired
+    CypherRepository cypherRepository;
+
+    @Autowired
+    ObjectGenerator generator;
+
+    @Test
+    public void findAllZeroTest() {
+        List<Progress> progressList = repository.findAll();
+
+        assertEquals(0, progressList.size());
+    }
+
+    @Test
+    public void addTest() {
+        Team team = generator.generateTeam();
+        teamRepository.save(team);
+
+        Cypher cypher = generator.generateCypher();
+        cypherRepository.save(cypher);
+
+        ProgressKey progressKey = new ProgressKey(cypher.getId(), team.getId());
+        Progress progress = new Progress(progressKey, team, cypher);
+
+        Progress progress1 = repository.save(progress);
+
+        assertEquals(progress, progress1);
+    }
+
+    /*
+     *  progress of one team on all stages
+     * */
+    @Test
+    public void getTeamsProgressTest() {
+        Team team = generator.generateTeam();
+        teamRepository.save(team);
+
+        List<Cypher> cyphers = generator.generateCypherList(ObjectGenerator.NUMBER_OF_CYPHERS);
+        cypherRepository.saveAll(cyphers);
+
+        for (Cypher cypher : cyphers) {
+            ProgressKey progressKey = new ProgressKey(cypher.getId(), team.getId());
+            repository.save(new Progress(progressKey, team, cypher));
+        }
+
+        assertEquals(ObjectGenerator.NUMBER_OF_CYPHERS, repository.count());
+    }
+
+    /*
+     * status of one cypher for all teams
+     * */
+    @Test
+    public void getStatusOfCypherTest() {
+
+        List<Team> teams = generator.generateTeamList();
+        teamRepository.saveAll(teams);
+
+        Cypher cypher = generator.generateCypher();
+        cypherRepository.save(cypher);
+
+        for (Team team : teams) {
+            ProgressKey progressKey = new ProgressKey(cypher.getId(), team.getId());
+            repository.save(new Progress(progressKey, team, cypher));
+        }
+
+        assertEquals(ObjectGenerator.NUMBER_OF_TEAMS, repository.count());
+    }
+
+    @Test
+    public void findAllTest() {
+        fillTable();
+
+        List<Progress> progressList = repository.findAll();
+
+        assertEquals(ObjectGenerator.NUMBER_OF_TEAMS * ObjectGenerator.NUMBER_OF_CYPHERS, progressList.size());
+    }
+
+    @Test
+    public void updateStatusTest() {
+        fillTable();
+
+        Cypher cypher = cypherRepository.findByStage(TESTED_STAGE);
+        Team team = teamRepository.findByName(TEAM_NAME + TESTED_TEAM);
+
+        Progress progress = repository.findByTeamAndCypher(team, cypher);
+
+        progress.setCypherStatus(SOLVED);
+        Progress progress1 = repository.save(progress);
+
+        assertEquals(SOLVED, progress1.getCypherStatus());
+    }
+
+    private void fillTable() {
```

Comment:
Metoda by mohla byt lip pojmenovana. Co se plni, kam se plni, atd. Z nazvu to vypada, ze se naplni jedna tabulka, ale pritom se plni vsechny tabulky. Zkus prosim lepe pojmenovat.